### ISSUE TITLE
Pagination needs some adjustments to make the component 100% accessible

### ISSUE THREAD

**Issue by marcoscv-work** [liferay/clay#5233]
## Context:

Pagination need some adjustments to be totally accesible.

![image](https://user-images.githubusercontent.com/7963804/205544648-6c2a6dc2-b9c6-41ec-b90b-be7a07a936fa.png)

Is a pattern/component that from a design point of view is offering a really good new behaviour that can help to reach all pages with a visual reduction of noise when there is a lot of pages, here are running out of the standard here so we will need to try something new, here a list of changes/improvements

- A pagination must be a nav so we we need to add <nav> container
- This <nav> needs an aria-label "Pagination" by default but it could be customised, for example if there is more than one pagination in the page, so please offer the possibility of changing the default aria-label
- We should build the nav by links, so please let's not use buttons, they should be removed from the examples
- Current page link should be present (sometimes is a link without href), aria-current="page" is enough to identify the active page in the <nav>
- Trigger for showing level-2 pagination buttons -> [...] Should have an aria-label and title like: "Show pages links from X to X". In this way we can understand the decision of skipping or showing that group of pages
- Title and aria-label for next/prev pages should be always in the link and provided by default, if not we were creating a double focus, first focus then in label:
![image](https://user-images.githubusercontent.com/7963804/205544404-20a25162-a25f-4a52-9710-f670f7404a0f.png) _(Markup from DXP, list of web content)_
- If we need to hide next/last page link, just change the link for a no focusable element like a span, no buttons with disable or links with layers, they will be accessible by rotor and buttons and links lists

### Warning about dropdown/level-2:

- There is not examples about dropdown menus inside a pagination, we are innovating a little bit here, but if we respect the composition of a nav element everything should be Ok
- We cannot use here a clay dropdown, we need to keep the level-2 structure of a navigation always in the page and exposed to accessibility tree, so the structure should be something like (pseudocode):
```
<nav
   <ul level-1
       <li page-1
           "trigger" aria-expanded="" aria-controls=""
          <ul  level 2
               <li page-2
               <li page-3 ...
```
 - Use of aria-expanded="" aria-controls="" is needed

**Comment by pat270** [comment]
> We should build the nav by links, so please let's not use buttons, they should be removed from the examples

Just curious why we shouldn't use buttons. If we are using javascript to swap out page content, wouldn't it be better to use a button instead of a link? If the `page-link` is the current page, we can swap the element to be a `span` or placeholder `a` tag.

**Comment by ethib137** [comment]
@marcoscv-work might have another reason, but my thought would be that usually with pagination we want the current page and delta to persist with a page refresh, so we usually update the url with this information. As a result, it is often implemented more like a same page navigation and should still be an anchor tag rather than a button.

**Comment by matuzalemsteles** [comment]
Yes that's true, normally in current applications that use hybrid rendering or router strategy on the client, paging like this is still done on the client but the tag will always be an `a` tag instead of button, what is done is that links that need to be handled by the router add click events to the `a` tag element to take away the browser's native behavior of browsing, but still control the history state of the browser.

**Comment by matuzalemsteles** [comment]
> We cannot use here a clay dropdown, we need to keep the level-2 structure of a navigation always in the page and exposed to accessibility tree, so the structure should be something like (pseudocode):

@marcoscv-work what do you mean by that? if I understand correctly the menu cannot be rendered in the body? should that be a problem? because in theory `aria-controls` or `aria-owns` would do exactly that for elements that can't be rendered following the order of the structure, I think doing that would bring us a lot more problems. Do you think we need something in our implementation today? because what we have now in Pagination is DropDown with accessibility and we are respecting level 2 even if it is rendered in the body but with `aria-controls`, we can also add `aria-owns` if necessary.

**Comment by marcoscv-work** [comment]
We could try go ahead with the dropdown, but it's really weird to have it in the middle of a pagination nav, more than just avoiding the alerts it's more about whether or not it makes sense.

We will review the component in Lexicon. :-D

**Comment by matuzalemsteles** [comment]
I see, wouldn't this be the same case for a [menubar with sub menus](https://www.w3.org/WAI/ARIA/apg/example-index/menubar/menubar-navigation.html)? maybe we can apply to that accessibility?

**Comment by marcoscv-work** [comment]
It could make sense, but for that we would need to change the navigation by tabkey by arrows, and it would make the pagination a little bit weird since we would not following the standard for a pagination (just links to pages)
